packages/react-spinner/src/components/Spinner/useSpinnerStyles.ts

```diff
@@ -1,34 +1,254 @@
+const spinnnerSizes = {
+  tiny: '20px',
+  extraSmall: '24px',
+  small: '28px',
+  medium: '32px',
+  large: '36px',
+  extraLarge: '40px',
+  huge: '44px',
 };
+
+// Stroke widths for the Spinner
+const spinnerStrokeWidth = {
+  sWidth: '2px',
+  mWidth: '3px',
+  lWidth: '4px',
+};
+
 /**
  * Styles for the root slot
  */
-const useStyles = makeStyles({
+const useRootStyles = makeStyles({
   root: {
-    // TODO Add default styles for the root element
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    ...shorthands.gap('8px'),
+  },
+
+  horizontal: {
+    flexDirection: 'row',
+  },
+
+  vertical: {
+    flexDirection: 'column',
+  },
+});
+
+const useLoaderStyles = makeStyles({
+  // global SVG class
+  spinnerSVG: {
+    [`& > svg`]: {
+      animationName: {
+        '0%': { transform: 'rotate(0deg)' },
+        '100%': { transform: 'rotate(360deg)' },
+      },
+      animationDuration: '3s',
+      animationIterationCount: 'infinite',
+      animationTimingFunction: 'linear',
+      backgroundColor: 'transparent',
+    },
+    [`& > svg > circle`]: {
+      cx: '50%',
+      cy: '50%',
+      fill: 'none',
+    },
+  },
+
+  tiny: {
+    [`& > svg`]: {
+      height: spinnnerSizes.tiny,
+      width: spinnnerSizes.tiny,
+    },
+    [`& > svg > circle`]: {
+      strokeWidth: spinnerStrokeWidth.sWidth,
+      r: rvalues.tiny,
+    },
   },
 
-  // TODO add additional classes for different states and/or slots
+  extraSmall: {
+    [`& > svg`]: {
+      height: spinnnerSizes.extraSmall,
+      width: spinnnerSizes.extraSmall,
+    },
+    [`& > svg > circle`]: {
+      strokeWidth: spinnerStrokeWidth.sWidth,
+      r: rvalues.extraSmall,
+    },
+  },
+
+  small: {
+    [`& > svg`]: {
+      height: spinnnerSizes.small,
+      width: spinnnerSizes.small,
+    },
+    [`& > svg > circle`]: {
+      strokeWidth: spinnerStrokeWidth.sWidth,
+      r: rvalues.small,
+    },
+  },
+
+  medium: {
+    [`& > svg`]: {
+      height: spinnnerSizes.medium,
+      width: spinnnerSizes.medium,
+    },
+    [`& > svg > circle`]: {
+      strokeWidth: spinnerStrokeWidth.mWidth,
+      r: rvalues.medium,
+    },
+  },
+
+  large: {
+    [`& > svg`]: {
+      height: spinnnerSizes.large,
+      width: spinnnerSizes.large,
+    },
+    [`& > svg > circle`]: {
+      strokeWidth: spinnerStrokeWidth.mWidth,
+      r: rvalues.large,
+    },
+  },
+
+  extraLarge: {
+    [`& > svg`]: {
+      height: spinnnerSizes.extraLarge,
+      width: spinnnerSizes.extraLarge,
+    },
+    [`& > svg > circle`]: {
+      strokeWidth: spinnerStrokeWidth.mWidth,
+      r: rvalues.extraLarge,
+    },
+  },
+
+  huge: {
+    [`& > svg`]: {
+      height: spinnnerSizes.huge,
+      width: spinnnerSizes.huge,
+    },
+    [`& > svg > circle`]: {
+      strokeWidth: spinnerStrokeWidth.lWidth,
+      r: rvalues.huge,
+    },
+  },
+
+  // global class for Spinner track
+  spinnerTrack: {
+    [`& > svg > circle#track`]: {
```

Comment:
We should not use `id`s as there may be multiple Spinners on a page and users may reasonably use an id like `track`. Using an id in a selector also makes styles harder to override because of their higher specificity.